src/ACEtk/block/details/BaseDistributionData.hpp

```diff
@@ -0,0 +1,193 @@
+#ifndef NJOY_ACETK_BLOCK_DETAILS_DISTRIBUTIONDATABASE
+#define NJOY_ACETK_BLOCK_DETAILS_DISTRIBUTIONDATABASE
+
+// system includes
+#include <variant>
+
+// other includes
+#include "utility/overload.hpp"
+#include "ACEtk/block/details/Base.hpp"
+#include "ACEtk/block/details/ColumnData.hpp"
+#include "ACEtk/block/InterpolationData.hpp"
+
+namespace njoy {
+namespace ACEtk {
+namespace block {
+namespace details {
+
+/**
+ *  @class
+ *  @brief Base class for distribution data as a function of incident energy
+ */
+template< typename Distribution >
+class BaseDistributionData : protected details::Base {
+
+  /* fields */
+  std::size_t locb_;
+  InterpolationData interpolation_;
+  details::ColumnData data_;
+
+  /* auxiliary functions */
+  #include "ACEtk/block/details/BaseDistributionData/src/generateXSS.hpp"
+  #include "ACEtk/block/details/BaseDistributionData/src/verifyIncidentEnergyIndex.hpp"
+  #include "ACEtk/block/details/BaseDistributionData/src/generateBlocks.hpp"
+  #include "ACEtk/block/details/BaseDistributionData/src/verifySize.hpp"
+
+public:
+
+  /* constructor */
+  #include "ACEtk/block/details/BaseDistributionData/src/ctor.hpp"
+
+  /**
+   *  @brief Return the interpolation data
+   */
+  auto interpolationData() const { return this->interpolation_; }
```

Comment:
[unsure]
Would it be better to return this as `auto &`?

Reply:
I changed this to const InterpolationData&

auto makes people lazy ...